src/formio/components/selectboxes.ts

```diff
@@ -0,0 +1,18 @@
+import {InputComponentSchema} from '..';
+
+type Validator = 'required';
+type TranslatableKeys = 'label' | 'description' | 'tooltip';
+
+export type SelectboxesInputSchema = InputComponentSchema<string, Validator, TranslatableKeys>;
+
+/**
+ * @group Form.io components
+ * @category Concrete types
+ */
+export interface SelectboxesComponentSchema
+  extends Omit<SelectboxesInputSchema, 'hideLabel' | 'disabled'> {
+  type: 'selectboxes';
+  // OF custom properties
+  dataSrc: 'manual' | 'variable';
+  itemsExpression?: string;
```

Comment:
I would stop there in Python, but is it possible to have a smarter definition in TS? If `dataSrc` is `variable`, `itemExpression` is required

Reply:
yeah so this one is pretty cool because you can use discriminated unions to properly narrow the types!

```ts
// to be shared with select and radio too
interface Option {
  value: string;
  label: string;
}

interface ManualValues {
  dataSrc: 'manual',
  values: Option[],
}

interface VariableValues {
  dataSrc: 'variable';
  itemsExpression: string | JsonObject;
}

type SelectboxesValues = ManualValues | VariableValues;

interface SelectboxesComponentSchema extends ..., SelectboxesValues {
  ...
}
```

with somewhere as a shared type:

```ts
type JsonValue = null | string | number | boolean | JsonArray | JsonObject;
type JsonArray = JsonValue[];
type JsonObject = Record<string, JSONValue>;
```

I've run into some annoying situations where we had to deal with/assign to json object from more specific types, but in this case I don't think it will cause problems since it's pretty much only the backend that actually uses `itemsExpression`.

Reply:
`SelectboxesComponentSchema` can't extend `SelectboxesValues` as it is a union type: https://stackoverflow.com/questions/56085306/error-message-an-interface-can-only-extend-an-object-type-or-intersection-of-o

Is it ok to make `SelectboxesComponentSchema` a `type`?